### DISCUSSION TITLE
What is an ideal way to get both front matter date and git commit for a page?

### DISCUSSION THREAD

**Discussion by saneef** [11ty/eleventy#2332] [Q&A] [answered]
The new Eleventy v1.0.1 comes with [`git Last Modified`](https://www.11ty.dev/docs/dates/#setting-a-content-date-in-front-matter) to get date from file's latest commit. 

Is there a clean way to access two dates – assigned date (as creation date) and `git Last Modified` (as last modified date) – of a page always?

Here are my use cases:
1. I want to show published date (a date provided through file name or front matter) and last update date (git commit date) for a blog post.
2. In RSS feeds, I would like to populate `<updated>` tag with the file's latest git commit date. 

For the case 2, currently I'm [using a plugin to get git commit date](https://saneef.com/tutorials/fix-dates-on-eleventy-rss-feeds/) to provide dates only for RSS feed.

**Comment by pdehaan** [comment, marked as answer]
I can really only think of two good solutions:
1. Create custom filters/shortcodes which get the file creation date or git modified from the OS. Although I'm not 100% confident that that will work in all cases (if Netlify or wherever you host does a fresh clone of a repo prior to deploying and potentially resets some dates).
2. Hardcode both dates in your front matter (or set the creation date via file name, and manually manage the updated date separately).

I always find "last modified" dates tricky. Just because I fixed some invalid HTML markup in a post doesn't mean the content was necessarily updated. So in some cases I think you're safer to manually set a last updated date manually when the content itself has been revised.

So maybe something like a ./src/blog/2022-04-11-hello-world.liquid file, which might have some front matter like:
```liquid
---
title: Hello World
last_updated: 2022-04-19
---

<h1>{{ title }}</h1>
```

Or even an array/changelog of changes

```liquid
---
title: Hello World
changelog:
	- date: 2022-04-19
	  message: Added more examples
    - date: 2022-04-13
      message: Fixed contact email address
---

<h1>{{ title }}</h1>
```

Might be a bit overkill in a lot of cases though.

**Reply by saneef** [reply]
Thanks @pdehaan for the ideas. 

I always forget to update the dates before commits. 😭 

I'll stick with current approach which is similar to your first suggestion. I'm using a filter (packed as [a plugin](https://github.com/saneef/eleventy-plugin-git-commit-date)).
